posthog/api/event_definition.py

```diff
@@ -27,4 +34,25 @@ class EventDefinitionViewSet(
     search_fields = ["name"]
 
     def get_queryset(self):
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
+            from ee.models.event_definition import EnterpriseEventDefinition
+
+            return self.filter_queryset_by_parents_lookups(EnterpriseEventDefinition.objects.all()).order_by(
+                self.ordering
+            )
         return self.filter_queryset_by_parents_lookups(EventDefinition.objects.all()).order_by(self.ordering)
+
+    def get_serializer_class(self) -> Type[serializers.ModelSerializer]:
+        serializer_class = self.serializer_class
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
+            from ee.api.enterprise_event_definition import EnterpriseEventDefinitionSerializer
+
+            serializer_class = EnterpriseEventDefinitionSerializer  # type: ignore
+        return serializer_class
+
+    def retrieve(self, request: Request, *args: Any, **kwargs: Any):
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
+            from ee.api.enterprise_event_definition import EnterpriseEventDefinitionSerializer
+
+            return response.Response(EnterpriseEventDefinitionSerializer(self.get_queryset().get(id=kwargs["id"])).data)
+        raise PermissionDenied("This is an Enterprise plan feature.")
```

Comment:
I don't think we should override this completely. Retrieving the object is still a valid operation for non-enterprise.